common/app/services/newsletters/GroupedNewslettersAgent.scala

```diff
@@ -0,0 +1,57 @@
+package services.newsletters
+
+import com.gu.Box
+import common.GuLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GroupedNewslettersAgent(newsletterApi: NewsletterApi) extends GuLogging {
+
+  val emptyGroupedNewslettersResponse = GroupedNewslettersResponse(
+    GroupedNewsletterResponse("News roundups", Nil),
+    GroupedNewsletterResponse("News by topic", Nil),
+    GroupedNewsletterResponse("Features", Nil),
+    GroupedNewsletterResponse("Fport", Nil),
+    GroupedNewsletterResponse("Culture", Nil),
+    GroupedNewsletterResponse("Lifestyle", Nil),
+    GroupedNewsletterResponse("Comment", Nil),
+    GroupedNewsletterResponse("Work", Nil),
+    GroupedNewsletterResponse("From the papers", Nil),
+  )
+
+  private val groupedNewslettersAgent =
+    Box[Either[String, GroupedNewslettersResponse]](Right(emptyGroupedNewslettersResponse))
+
+  def refresh()(implicit ec: ExecutionContext): Future[Either[String, GroupedNewslettersResponse]] = {
+    log.info("Refreshing Grouped Newsletters for round up page.")
+
+    val groupedNewslettersQuery = newsletterApi.getGroupedNewsletters()
```

Comment:
I'm wondering whether it'd be nicer if there were a single agent that either provided the data in 2 formats, or a single format that other places transformed as needed. I suppose that doesn't necessarily align with the Newsletters API, but I'm not immediately sure why there is a separate endpoint for grouped newsletters, and not a single one that returned a `group` property. Likely I'm missing something, just a thought, but probably one considered elsewhere previously.

Reply:
Single agent that makes 2 API requests and populates 2 boxes might be a minor simplification too as logically I think they belong together.

Reply:
I actually did that at first and I think I got confused about what were methods on a trait and what were methods I'd named. I'll go back in now and take a closer look now that I've got it working